### PR TITLE
Added cli and updated tail of first blog

### PR DESCRIPTION
**Pull request by sverhoeven**

### CODE REVIEW

File: js-webapp/README.md [hunk 1]
```diff
@@ -127,6 +126,43 @@ stored as the private member ``tolerance``. Once the object instance has been co
...
+To make sure the code actually works in C++ we will wrap it in a main function, compile it and run it.
+
+```cpp
+#include <iostream>
+#include <iomanip>
+
+#include "newtonraphson.hpp"
+
+int main() {
+  double initial_guess = -20;
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
  double initial_guess = -4;
```

File: js-webapp/README.md [hunk 2]
```diff
@@ -127,6 +126,43 @@ stored as the private member ``tolerance``. Once the object instance has been co
...
+  std::cout << std::fixed << std::setprecision(2) << x1 << std::endl;
+  return 0;
+}
+```
+File: _cli.cpp_
+
+Compile it
+
+```shell
+c++ -o newtonraphson -I. newtonraphson.cpp cli.cpp
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
g++ -o newtonraphson -I. newtonraphson.cpp cli.cpp
```

File: js-webapp/README.md [hunk 3]
```diff
@@ -127,6 +126,43 @@ stored as the private member ``tolerance``. Once the object instance has been co
...
+And run it, to see if it returns the correct answer, `-1.00` in our case.
+
+```shell
+./newtonraphson
+The value of the root is : -1.00
+```
+
+Nice, it works as expected.
+
+Next let's define the JavaScript interface our class should have with Emscripten bindings.
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
Next, let's define the JavaScript interface with Emscripten bindings.
```

File: js-webapp/README.md [hunk 4]
```diff
@@ -188,7 +224,7 @@ following HTML:
             const newtonraphson = new NewtonRaphson(tolerance);
             const root = newtonraphson.solve(initial_guess);
             // Write root to tag with answer as identifier
-            document.getElementById("answer").innerHTML = root.toFixed(2);
+            document.getElementById("answer").innerHTML = root.toFixed(4);
```
Comment (fdiblen) [inline js-webapp/README.md]: The cli has a precision of 2. Why do we have 4 here?

File: js-webapp/README.md [hunk 5]
```diff
@@ -212,18 +248,14 @@ Visit [http://localhost:8000/](http://localhost:8000/) to see the result of the
 
 _The resulting page if everything works._
 
-**TODO** (Recap and announce what else is coming)
+Looking at the plot of the equation we expect the root to be `-1.00`, which is exactly what the HTML page in the browser is showing.
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
Looking at the plot of the equation we expect the root of the equation to be `-1.00`, which is exactly what the HTML page in the browser is showing.
```

File: js-webapp/README.md [hunk 6]
```diff
@@ -212,18 +248,14 @@ Visit [http://localhost:8000/](http://localhost:8000/) to see the result of the
...
 
-The result of root finding was calculated using the C++ algorithm compiled to a WebAssembly module, executed by some
-JavaScript and rendered on a HTML page.
+So what did we do, we wrote a simple algorithm in C++, wrote Emscripten bindings, compiled a WebAssembly module and ran the algorithm in a web browser using some JavaScript to talk to the module.
```
Comment (fdiblen) [inline js-webapp/README.md]: Can we rephrase it? It is difficult to read. May be it helps to split it to multiple sentences.

File: js-webapp/README.md [hunk 7]
```diff
@@ -212,18 +248,14 @@ Visit [http://localhost:8000/](http://localhost:8000/) to see the result of the
...
 
 The nice thing about this solution is that we don't need expensive infrastructure to perform computation as the
 computation is done in the users web browser. We just need somewhere to host the files.
 
-<!--
-In upcoming blogs will see if we can perform the computation without blocking the user interface and make a nice
-interactive form. In even more blogs we will look into performing the computation on the server with JavaScript and
-Python in a human readable and compute readable format.
--->
+In upcoming blogs will see how we can perform the computation without blocking the user interface, make a nice
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
In upcoming blogs we will see how we can perform the computation without blocking the user interface, make a nice
```

File: js-webapp/README.md (hunk 4 above)
Comment (sverhoeven) [reply js-webapp/README.md]: Doh, you are correct, I wanted to compromise and go to 4, but changed my mind halfway.

File: js-webapp/README.md [hunk 8]
```diff
@@ -212,18 +248,14 @@ Visit [http://localhost:8000/](http://localhost:8000/) to see the result of the
...
+interactive form and make a visualization using a scatter plot. In a wrap up blog we will combine all the code of this and upcoming JavaScript WebAssembly blogs in a final web application.
```
Comment (fdiblen) [inline js-webapp/README.md]:
```suggestion
interactive form and make a visualization to show data from each iteration. In a wrap up blog we will combine all the code of this and upcoming JavaScript WebAssembly blogs in a final web application.
```

File: js-webapp/cli.cpp [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+#include <iomanip>
+
+#include "newtonraphson.hpp"
+
+int main() {
+  double initial_guess = -20;
```
Comment (fdiblen) [inline js-webapp/cli.cpp]:
```suggestion
  double initial_guess = -4;
```

**Review by fdiblen — CHANGES_REQUESTED** [review]
cliclicli
:flamingo: 

Please see my comments.

File: js-webapp/README.md (hunk 6 above)
Comment (sverhoeven) [reply js-webapp/README.md]: rewrote to bullet list

ps. also shortened cli section.

File: js-webapp/README.md (hunk 2 above)
Comment (sverhoeven) [reply js-webapp/README.md]: Moved compile command (g++) to cli.cpp

**Comment by jspaaks** [comment]
Continued in #31, closing this one

**Pull request closed without merging**